tools/inventory-extract-windows/main.go

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/diskfs/go-diskfs/filesystem/fat32"
+	"github.com/diskfs/go-diskfs/partition"
+)
+
+var dirFlag = flag.String("dir", "", "save file to provided directory (will generate temp if empty)")
+var driveNumber = flag.Uint("drive", 0, "open drive with provided number")
+var showDrives = flag.Bool("list", false, "list drives")
+var helpFlag = flag.Bool("help", false, "help")
+
+func main() {
+	flag.Parse()
+	if *helpFlag {
+		fmt.Println("This program do export of INVENTORY partition from EVE drive")
+		fmt.Println("You can see list of available drives with -list flag")
+		fmt.Println("Then you must set Index of drive to -drive flag")
+		fmt.Println("With -dir flag you can provide directory to save files into. " +
+			"The program will create directory in temp if you will not provide this flag.")
+		return
+	}
+	if *driveNumber == 0 {
+		fmt.Println("Please provide -drive flag with Index of device from the list below:")
+		*showDrives = true
+	}
+	if *showDrives {
+		res, err := exec.Command("wmic", "diskdrive",
+			"where", "MediaType='Removable Media'",
+			"get", "index,caption").CombinedOutput()
+		if err != nil {
+			log.Println(err)
+		} else {
+			fmt.Println(string(res))
+		}
+		time.Sleep(time.Second * 5)
+		return
+	}
+	_, err := os.Open(`\\.\PhysicalDrive0`)
+	if err != nil {
+		log.Println("This program must be run by Administrator")
+		log.Println("We will ask for permissions")
+		time.Sleep(time.Second * 5)
+		runMeElevated()
+		os.Exit(0)
+	}
+	f, err := os.Open(fmt.Sprintf(`\\.\PhysicalDrive%d`, *driveNumber))
+	if err != nil {
+		log.Println(err)
+		time.Sleep(time.Second * 5)
+		os.Exit(1)
+	}
+	defer f.Close()
+
+	table, err := partition.Read(f, 512, 512)
+
+	if err != nil {
+		log.Println(err)
+		time.Sleep(time.Second * 5)
+		os.Exit(1)
+	}
+	partitions := table.GetPartitions()
+	found := false
+	for _, el := range partitions {
+		fsCurrent, err := fat32.Read(f, el.GetSize(), el.GetStart(), 512)
+		if err != nil {
+			continue
+		}
+		if fsCurrent.Label() == "INVENTORY" {
+			found = true
+			var dir string
+			if *dirFlag != "" {
+				dir, err = filepath.Abs(*dirFlag)
+				if err != nil {
+					log.Println(err)
+					continue
+				}
+				err = os.MkdirAll(dir, os.ModeDir)
```

Comment:
is a warning needed if the path already exists. Should we nuke the old directory?

Reply:
No, we want to save new portions of data here, in the same directory. And search next by modification time

Reply:
I guess we also ok with rewriting pre-existing data silently then?